src/v/cluster/members_frontend.cc

```diff
@@ -132,4 +138,81 @@ members_frontend::recommission_node(model::node_id id) {
     }
     co_return res.value().data.error;
 }
+
+ss::future<members_frontend::get_under_replicated_reply>
+members_frontend::get_under_replicated(model::node_id id) {
+    constexpr model::node_id no_leader{-1};
+    auto rep = co_await get_local_under_replicated();
+    if (id != _self() || rep.under_replicated) {
```

Comment:
What situations would produce `id != _self`?  It seems we're always querying our own under-replicatedness.

Reply:
> What situations would produce `id != _self`? It seems we're always querying our own under-replicatedness.

E.g.: Node 1 wants to know if it's under_replicated
It checks itself to see if it's the leader of any partition, and if so, whether it's under_replicated. In hindsight, I don't think this is possible.
Then it sends a message to Node 0 and Node 2, to ask them if it's under replicated, they know, iff they're the leader.